pkg/interop/util/util.go

```diff
@@ -22,6 +26,34 @@ func FromAddress(address string) interop.Hash160 {
 	return nil
 }
 
+// AddressToHash160 is a utility function that converts a Neo address to its hash
+// (160 bit BE value in a 20 byte slice). Unlike the FromAddress, AddressToHash160
+// will generate the honest Neo-VM script for runtime address conversion, thus
+// AddressToHash160 may be used for strings got at runtime. For strings those value
+// is known at compilation time, use FromAddress for the sake of lower script cost.
```

Comment:
We can optimize this call too for static strings. And even deprecate `util.FromAddress`.

Reply:
Fixed with a bit of compiler magic, but this case is rather tricky, because `address.ToHash160` is dynamic builtin function (we have only static ones before). Its "dynamycness" is defined by the type of argument, so we have to decide whether we need to inline it during compilation time for each separate case.

Also, I've discovered that all our interop functions either are being inlined or the code generation for them is handled manually, so there's no way to handle interop function as a standard function call and call it afterwards via `CALL` instruction. That means that interop functions (except for `util.FromAddress` and `neogointernal`) will be inlined each time they're used which is OK for small functions but can be a problem for `address.ToHash160`. Its code is not so trivial, thus repeating it again and again leads to a large program size. Here's the code of inlined `address.ToHash160`:
```
43       LDSFLD0      
44       DUP          
45       ISTYPE       Buffer (30)
47       JMPIF        51 (4/04)
49       CONVERT      Buffer (30)
51       STLOC1       
52       LDLOC1       
53       CALLT        0000 ("\x00\x00")
56       STLOC0       
57       LDLOC0       
58       DUP          
59       ISNULL       
60       JMPIF        65 (5/05)
62       SIZE         
63       JMP          67 (4/04)
65       DROP         
66       PUSH0        
67       PUSHINT8     21 (15)
69       JMPEQ        96 (27/1b)
71       PUSHDATA1    696e76616c69642061646472657373206c656e677468 ("invalid address length")
95       THROW        
96       LDLOC0       
97       PUSH0        
98       PICKITEM     
99       SYSCALL      System.Runtime.GetAddressVersion (4c4992dc)
104      JMPEQ        131 (27/1b)
106      PUSHDATA1    696e76616c6964206164647265737320707265666978 ("invalid address prefix")
130      THROW        
131      LDLOC0       
132      PUSH1        
133      PUSHINT8     21 (15)
135      OVER         
136      SUB          
137      SUBSTR       
```

Reply:
That can be a tricky part, I've tried to solve it immediately, but a lot of tests are failing, so this needs a separate investigation, see https://github.com/nspcc-dev/neo-go/issues/2700.

Currently the following optimisation is implemented: `address.ToHash160` acts exactly like `util.FromAddress` if the argument is BasicLiteral (constant string). If not, then the code of `address.ToHash160` is being inlined each time it's used.

Reply:
`util.FromAddress` is deprecated, we can remove it later with a few lines of related code.

Reply:
It's OK to inline this chunk, I don't think we'll have a lot of uses of it in the same contract, usually it's one-two places that need this.